services/worker/src/worker/job_runners/config/opt_in_out_urls_count.py

```diff
@@ -0,0 +1,121 @@
+# SPDX-License-Identifier: Apache-2.0
+# Copyright 2022 The HuggingFace Authors.
+
+import logging
+from http import HTTPStatus
+from typing import Any, Literal, Mapping, Optional, Tuple
+
+from libcommon.constants import PROCESSING_STEP_CONFIG_OPT_IN_OUT_URLS_COUNT_VERSION
+from libcommon.simple_cache import DoesNotExist, SplitFullName, get_response
+
+from worker.job_runner import (
+    JobResult,
+    JobRunner,
+    JobRunnerError,
+    ParameterMissingError,
+    get_previous_step_or_raise,
+)
+from worker.utils import OptInOutUrlsCountResponse
+
+ConfigOptInOutUrlsCountJobRunnerErrorCode = Literal["PreviousStepFormatError"]
+
+
+class ConfigOptInOutUrlsCountJobRunnerError(JobRunnerError):
+    """Base class for exceptions in this module."""
+
+    def __init__(
+        self,
+        message: str,
+        status_code: HTTPStatus,
+        code: ConfigOptInOutUrlsCountJobRunnerErrorCode,
+        cause: Optional[BaseException] = None,
+        disclose_cause: bool = False,
+    ):
+        super().__init__(
+            message=message, status_code=status_code, code=code, cause=cause, disclose_cause=disclose_cause
+        )
+
+
+class PreviousStepFormatError(ConfigOptInOutUrlsCountJobRunnerError):
+    """Raised when the content of the previous step has not the expected format."""
+
+    def __init__(self, message: str, cause: Optional[BaseException] = None):
+        super().__init__(message, HTTPStatus.INTERNAL_SERVER_ERROR, "PreviousStepFormatError", cause, False)
+
+
+def compute_opt_in_out_urls_scan_response(dataset: str, config: str) -> Tuple[OptInOutUrlsCountResponse, float]:
+    logging.info(f"get config-opt-in-out-urls-count for dataset={dataset} config={config}")
+
+    split_names_response = get_previous_step_or_raise(
+        kinds=["/split-names-from-streaming"], dataset=dataset, config=config
+    )
+    content = split_names_response.response["content"]
+    if "splits" not in content:
+        raise PreviousStepFormatError("Previous step did not return the expected content: 'splits'.")
+
+    urls_columns = []
+    num_opt_in_urls = 0
+    num_opt_out_urls = 0
+    num_urls = 0
+    num_scanned_rows = 0
+    try:
+        total = 0
+        pending = 0
+        for split_item in content["splits"]:
+            split = split_item["split"]
+            total += 1
+            try:
+                response = get_response(kind="split-opt-in-out-urls-scan", dataset=dataset, config=config, split=split)
```

Comment:
maybe use `split-opt-in-out-urls-count` instead?

Reply:
Yes sure, I forgot to change this one but it was fixed on https://github.com/huggingface/datasets-server/pull/1108/commits/f142efba4f815c1af9b654eb3bdb346dd501b9a7